chain/consensus/mir/membership/membership.go

```diff
@@ -11,6 +13,41 @@ import (
 	"github.com/filecoin-project/lotus/chain/ipcagent/rpc"
 )
 
+type Source int32
+
+const (
+	FakeSource    Source = 0
+	StringSource  Source = 1
+	FileSource    Source = 2
+	OnChainSource Source = 3
+)
+
+func (e Source) String() string {
```

Comment:
I don't see where this method is used (but I - mostly my IDE - might have missed it).